src/components/TextField/tests/TextField.test.tsx

```diff
@@ -305,6 +305,41 @@ describe('<TextField />', () => {
     });
   });
 
+  describe('characterCount', () => {
+    it('counts characters in input field', () => {
+      const textField = mountWithAppProvider(
+        <TextField
+          value="test"
+          showCharacterCount
+          label="TextField"
+          id="MyField"
+          onChange={noop}
+        />,
+      );
+
+      const characterCount = textField.find('#MyField-character-counter');
```

Comment:
This is probably a good case for using `findByTestID`. Throw a `testID` on the div element and
```suggestion
       const characterCount = findByTestID(textField, 'CharacterCounter');
```